sources/imdsclient/src/lib.rs

```diff
@@ -30,25 +30,6 @@ pub struct ImdsClient {
     session_token: String,
 }
 
-/// This is the return type when querying for the IMDS identity document, which contains information
```

Comment:
- Both caller sites are using one-or-the-other of the fields.
- instance-identity-document can be absent under certain circumstances.
- Before the refactor instance type was coming from meta-data and not instance-identity-document.

Adding this all up, it seemed better to go back to the behavior that was present before the refactor. The other alternative is to return an Option<InstanceIdentityDocument> which has Option<String> fields which seems slightly less convenient and could fail to get the instance_type in cases that we previously didn't fail to get the identity_type.  🤷 

So we get_region back to metadata (which is where it was coming from before) and kept get_instance_type as a call

Reply:
Sounds good to me!  Thanks for that.

Reply:
I'm going to condense what @webern said above into the commit message as well as test additional variants without user-data. This is a little light on detail and testing (hence posted as draft) in its current form, but I wanted to make it available for eyes sooner rather than later.